docs/README.md

```diff
@@ -900,15 +900,145 @@ where the schema for `CreateProductInput` is
 
 ### 2. Events
 
+Events are **immutable records of facts** within your application's domain. They are the cornerstone of Booster because of its event-driven and event-sourced nature. Booster events are TypeScript classes decorated with `@Event`. An event class may look like this:
+
+```typescript
+@Event
+export class EventName {
+  public constructor(readonly field1: SomeType,
+                     readonly field2: SomeOtherType,
+                     /* This event's entity ID must be present */) {}
+
+  public entityID(): UUID {
+    return /* the associated entity ID */
+  }
+}
+```
+
+Events and [entities](#4-entities-and-reducers) are intimately related. All events belong to one entity through the `entityID` method, and entities represent the application's state after reducing the stream of events. Indeed, an entity is just an aggregated representation of the same data present in its events. It is possible to rebuild entities from events at any time. Booster guarantees that all the events associated with an entity, will be reduced in the same order they were stored. Take a look at this event:
+
+```typescript
+@Event
+export class CartPaid {
+  public constructor(
+    readonly cartID: UUID,
+    readonly paymentID: UUID) {}
+
+  public entityID(): UUID {
+    // returns cartID because we want to associate
+    // (and reduce) it within the Cart entity
+    return this.cartID
+  }
+}
+```
+
+An event must be able to find the entity ID value within the event attributes. Either among a parameter injected in the constructor or because it is nested in one of them. In the `CartPaid` example, the entity ID (`paymentID`) is injected directly. However, there are scenarios where the entity ID is not known, or it doesn't exist. For example, in events like `ProductCreated` you can return a completely new ID:
```

Comment:
I see that you left this as it was. Didn't you agree with my suggestion? ([this one](https://github.com/boostercloud/booster/pull/226#discussion_r449045139)

Reply:
I'm sorry, I think I forgot to rephrase this, I will. In fact, I included the new UUID example for the case of the `Created*` like events because that helps a lot. The singleton entity use case is also interesting.